.github/workflows/dockerfiles.yaml

```diff
@@ -0,0 +1,30 @@
+name: Update Dockerfiles on Default Branch Merge
+
+on:
+  pull_request:
+    types: [closed]
+
+jobs:
+  update-dockerfiles:
+    name: Update Dockerfiles on Default Branch Merge
+    runs-on: ubuntu-latest
+    container: python:3.8
+    if: github.event.pull_request.base.ref == github.event.repository.default_branch
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2.1.1
+      - name: Generate new Dockerfiles
+        run: |
+        rm -rf generated-dockerfiles/*.Dockerfile
+        pip install Jinja2 PyYAML
+        python generate_dockerfiles.py
+        mv build/* generated-dockerfiles/
+      - name: Setup git Config
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+      - name: Commit to Repository
+        run: |
+          git add generated-dockerfiles/*
+          git commit -m 'update dockerfiles'
```

Comment:
Automated commits should have a more descriptive response. Maybe something like

`[GH Actions] Generate Dockerfiles for PR #XX`